### PR TITLE
Exclude versions with due date from Needs Human Review queue

### LINKED ISSUE #20279: Exclude versions with a due date from the Needs human review queue
With https://github.com/mozilla/addons-server/pull/20252 when a `needs_human_review` flag is set, it will set a due date, so in the future versions needing human review will just be in the manual review queue. Existing versions however, don't have a due date and we decided to not do a migration for those and instead have reviewers clear the existing needs human review queue. To help them do that we should temporarily add a filter to exclude versions with a due date from the needs human review queue.

Once the queue is cleared, we'll then remove it completely in https://github.com/mozilla/addons-server/issues/20238.

### PR DESCRIPTION
**Pull request by wagnerand**
Fix #20279.

### CODE REVIEW

**Review by bobsilverberg — APPROVED** [review]
The code looks fine to me, but I have not really been keeping on top of all of these conversations about these requirements, so perhaps @diox or @eviljeff should take a look as well?

File: src/olympia/addons/tests/test_models.py
Comment (diox) [inline src/olympia/addons/tests/test_models.py:3182]: If you add an assertion I think you'll find that it does have a due date - because the `needs_human_review` forces it. You'd have to explicitly pass `'due_date': None` for that not to happen. I'm surprised the test is passing...

Comment (diox) [reply src/olympia/addons/tests/test_models.py:3182]: Your branch is missing https://github.com/mozilla/addons-server/commit/d960b4e3a7e0044e8d8559691ef84019dc3dc107 which has the logic to set `due_date` if `needs_human_review` is also set, which is why your test is passing despite not doing what I suggest. I recommend updating it with `master` and fixing the test before merging, just in case.

Comment (wagnerand) [reply src/olympia/addons/tests/test_models.py:3182]: Is that because of https://github.com/mozilla/addons-server/blob/60fdf84e050974524443979145ca6ea20a0ea362/src/olympia/versions/models.py#L1081-L1090? I am happy to update the test, the fact that it is passing without it is concerning? How can we make sure we're testing the right thing?

Comment (wagnerand) [reply src/olympia/addons/tests/test_models.py:3182]: Yep, that was it, and adding an explicit `None` due date makes it pass. Thanks!

**Comment by wagnerand** [comment]
hm, and this now unhid a bunch of other failures :(

**Review by diox — APPROVED** [review]

**Pull request merged**
